File: Online Judges/LeetCode/June LeetCoding Challenge/Week-3/Permutation Sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        numStr = [str(i) for i in range(1, n+1)]
        fact = [1]*n
        for i in range(1, n):
            fact[i] = i*fact[i-1]
        ans = []
        k -= 1
        while n:
            i = int(k / fact[n-1])
            k %= fact[n-1]
            ans.append(numStr[i])
            numStr.pop(i)
            n -= 1
        return ''.join(ans)

File: Online Judges/LeetCode/June LeetCoding Challenge/Week-3/Permutation Sequence.rs
impl Solution {
    pub fn get_permutation(n: i32, k: i32) -> String {
        let mut numStr: Vec<char> = (0..n as u8).map(|a|(b'1' + a) as char).collect();
        let mut k:i32 = k-1;
        let mut factN:i32 = (1..n).product();
        
        let mut Ans: String = String::with_capacity(n as usize);
        for i in 0..n-1{
            let id:usize = (k/factN) as usize;
            k %= factN;
            Ans.push(numStr[id]);
            numStr.remove(id);
            factN = factN / (n - 1 - i)
        }
        Ans.push(numStr[0]);
        return Ans;
    }
}